packages/pharos/src/components/table/pharos-table.ts

```diff
@@ -105,6 +118,41 @@ export class PharosTable extends ScopedRegistryMixin(PharosElement) {
     this.totalResults = !this.totalResults ? this.rowData.length : this.totalResults;
   }
 
+  private _initHeaderObserver(): void {
+    const ACTIVE_HEADER_CLASS = 'table-header--has-active-sticky-header';
+    if (!this.hasStickyHeader) {
+      throw new Error('Sticky header is not enabled, cannot initialize observer');
+    }
```

Comment:
Where does this condition arise in practice? It feels overly defensive, but I might be missing something.

Reply:
Yeah, that's probably fair, got a little carried away, i can take it out